tests/v2/validation/provisioning/provisioning_node_driver_test.go

```diff
@@ -403,163 +242,25 @@ func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioning_RKE2AWSClusterDyn
-			"worker":       true,
-		},
-	}
-
-	nodeRoles1 := []map[string]bool{
-		{
-			"controlplane": true,
-			"etcd":         false,
-			"worker":       false,
-		},
-		{
-			"controlplane": false,
-			"etcd":         true,
-			"worker":       false,
-		},
-		{
-			"controlplane": false,
-			"etcd":         false,
-			"worker":       true,
-		},
-	}
-
-	tests := []struct {
-		name      string
-		nodeRoles []map[string]bool
-		client    *rancher.Client
-	}{
-		{"1 Node all roles Admin User", nodeRoles0, r.client},
-		{"1 Node all roles Standard User", nodeRoles0, r.standardUserClient},
-		{"3 nodes - 1 role per node Admin User", nodeRoles1, r.client},
-		{"3 nodes - 1 role per node Standard User", nodeRoles1, r.standardUserClient},
-	}
-
-	var name string
-	for _, tt := range tests {
-		for _, kubeVersion := range r.kubernetesVersions {
-			name = tt.name + " Kubernetes version: " + kubeVersion
-			for _, cni := range r.cnis {
-				name += " cni: " + cni
-				r.Run(name, func() {
-					testSession := session.NewSession(r.T())
-					defer testSession.Cleanup()
-
-					testSessionClient, err := tt.client.WithSession(testSession)
-					require.NoError(r.T(), err)
-
-					clusterName := AppendRandomString(baseHarvesterClusterName)
-					generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
-					machinePoolConfig := machinepools.NewHarvesterMachineConfig(generatedPoolName, namespace)
-
-					machineConfigResp, err := machinepools.CreateMachineConfig(machinepools.HarvesterResourceConfig, machinePoolConfig, testSessionClient)
-					require.NoError(r.T(), err)
-
-					machinePools := machinepools.RKEMachinePoolSetup(tt.nodeRoles, machineConfigResp)
-
-					cluster := clusters.NewRKE2ClusterConfig(clusterName, namespace, cni, cloudCredential.ID, kubeVersion, machinePools)
-
-					clusterResp, err := clusters.CreateRKE2Cluster(testSessionClient, cluster)
-					require.NoError(r.T(), err)
-
-					result, err := testSessionClient.Provisioning.Clusters(namespace).Watch(context.TODO(), metav1.ListOptions{
-						FieldSelector:  "metadata.name=" + clusterName,
-						TimeoutSeconds: &defaults.WatchTimeoutSeconds,
-					})
-					require.NoError(r.T(), err)
-
-					checkFunc := clusters.IsProvisioningClusterReady
-
-					err = wait.WatchWait(result, checkFunc)
-					assert.NoError(r.T(), err)
-					assert.Equal(r.T(), clusterName, clusterResp.Name)
-
-				})
-			}
-		}
+func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioning() {
+	for provider_name := range r.providers {
+		fmt.Printf("Starting test for %v\n", r.providers[provider_name])
+		provider_struct := CreateProvider(r.providers[provider_name])
+		r.Provisioning_RKE2Cluster(provider_struct)
 	}
 }
 
-func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioning_RKE2HarvesterClusterDynamicInput() {
+func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioningDynamicInput() {
 	nodesAndRoles := NodesAndRolesInput()
+	fmt.Printf("%v", nodesAndRoles)
 	if len(nodesAndRoles) == 0 {
```

Comment:
Why do you have this twice. Do it either here, or in `Provisioning_RKE2ClusterDynamicInput`. If you do it here, then maybe `Provisioning_RKE2ClusterDynamicInput` Takes the nodeConfig Input as a parameter. Pick a place.

Reply:
I've removed all of the print statements I had in for now.  I can add some back in later if we decide we need them. Maybe we could have a meeting around when is appropriate to add logging?

Reply:
Sure, about the logging. It was more of a question I'm not against having all these log statements. Just not print statements.

Other note you have 

```
nodesAndRoles := NodesAndRolesInput()
if len(nodesAndRoles) == 0 {
 	r.T().Skip()
}
```

in two spots. I would prefer to just have at the top of `TestProvisioningDynamicInput`

Reply:
oh, I thought I'd removed that. Just updated the PR